plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -555,28 +554,34 @@ private class DokkaDescriptorVisitor(
         getDocumentation()?.toSourceSetDependent() ?: emptyMap()
 
     private fun ClassDescriptor.resolveClassDescriptionData(): ClassInfo {
-        tailrec fun buildInheritanceInformation(
-            inheritorClass: ClassDescriptor?,
-            interfaces: List<ClassDescriptor>,
+
+        fun toTypeConstructor(kt: KotlinType) =
+            TypeConstructor(DRI.from(kt.constructor.declarationDescriptor as DeclarationDescriptor), kt.arguments.map { it.toProjection() })
+
+        tailrec fun buildAncestryInformation(
```

Comment:
I know this was not introduced with this PR,; It still seems a little bit strange to me, why you are trying to represent a tree like structure within a flat `Set`? 🤔

Reply:
As far as I know it was introduced by @MarcinAman. I guess the clue is we don't need to preserve the relations between two given ancestors, we focus only on distance from given node. Therefore, the Set with `level` property is much more convenient, since you don't have to write your own abstraction and add utils for e. g. traversing the structure and collecting certain nodes. It can be rewritten, however I don't know whether it is the crucial task for now.